src/core/Akka.Streams/Dsl/RestartSink.cs

```diff
@@ -0,0 +1,138 @@
+﻿// //-----------------------------------------------------------------------
+// // <copyright file="RestartSink.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2021 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2021 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using Akka.Pattern;
+using Akka.Streams.Stage;
+
+namespace Akka.Streams.Dsl
+{
+    /// <summary>
+    /// A RestartSink wraps a <see cref="Sink"/> that gets restarted when it completes or fails.
```

Comment:
When you say "completes" here - you don't mean that when the graph completes right? That wasn't clear from reading this.

Reply:
No idea. I only seperated the Sink and Source code to seperate files at this point. Still need to go through them. Ill take a closer look at the comments then as well.

Reply:
The docs are the same as on the scala side. I think what is being referenced here is the stage itself. And not the entire graph.